### ISSUE TITLE
Process recording doesn't complete

### ISSUE THREAD

**Issue by kgilpin** [getappmap/appmap-node#118]
Running AppMap index service:

```
npx appmap-node env APPLAND_API_KEY=<key> node ./built/cli.js index -d ~/source/appland/appland -v -w -p 30101 
```

When the process is terminated via Ctrl-C, the `part` file is left behind and not completed as an appmap.json.

Branch you can use: https://github.com/getappmap/appmap-js/pull/1603

You can trigger a request to the service like this:

```
node ./bin/rpcClient explain '{"question":"Add a captcha to the user login" }'
```

Or

```
./bin/rpcClient search '{"query":"maxresults up-to-date parameters settings config maximumresults limit current updated latest","maxResults":10}'
```

Use your imagination :-)

**Comment by dividedmind** [comment]
Note this can be reproduced on main branch of appmap-js:
```console
$ appmap-node built/cli.js index -w -d /some/project  # works
$ appmap-node node built/cli.js index -w -d /some/project  # breaks
```

**Comment by zermelo-wisen** [comment]
I found that this bug occurs when there is a script with active `setInterval` call and `appmap-node` runs this script indirectly with a `node` process: `appmap-node node interval.js`.

Changing the way we forward signals (SIGINT resulting from Ctrl-C) to the child process seemed to solve the issue. I put a test that tests exactly this behavior and it fails without the fix.

Beside this, I tested the fix with the case reported by @dividedmind above: `appmap-node node built/cli.js index -w -d /some/project  # breaks`, assuming it's similar to the original command above which needs an APPLAND_API_KEY.
